### ISSUE TITLE
Unable to stop instance 

### ISSUE THREAD

**Issue by shalabhsingh** [jcjorel/ec2-spot-converter#10]
When I run the final command to convert an on demand instance to spot instance, the process is not going beyond step 6.

```[ec2-user@ip-172-31-43-48 ~]$ ./ec2-spot-converter --stop-instance --review-conversion-result --instance-id i-00df331228ea92843
[INFO] 2021-07-03 01:05:59,943 ec2-spot-converter - [STEP 1/26] Read DynamoDB state table...
[INFO] 2021-07-03 01:05:59,980 ec2-spot-converter -   => SUCCESS. Record 'i-00df331228ea82943' read succesfully.
[INFO] 2021-07-03 01:06:00,104 ec2-spot-converter - [STEP 2/26] Discover instance state... : RECOVERED STATE. SKIPPED!
[INFO] 2021-07-03 01:06:00,104 ec2-spot-converter -   => SUCCESS. Instance is in state running...
[INFO] 2021-07-03 01:06:00,105 ec2-spot-converter - [STEP 3/26] Deregister from ELB target groups... => SKIPPED! Need '--check-targetgroups' argument.
[INFO] 2021-07-03 01:06:00,105 ec2-spot-converter - [STEP 4/26] Wait for drainage of ELB target groups... => SKIPPED! Need '--check-targetgroups' argument.
[INFO] 2021-07-03 01:06:00,105 ec2-spot-converter - [STEP 5/26] Stop the instance... : RECOVERED STATE. SKIPPED!
[INFO] 2021-07-03 01:06:00,105 ec2-spot-converter -   => SUCCESS. Stopping 'i-00df331228ea82943'...
[INFO] 2021-07-03 01:06:00,105 ec2-spot-converter - [STEP 6/26] Wait for expected instance state...
[INFO] 2021-07-03 01:06:00,168 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:06:15,220 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:06:30,276 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:06:45,325 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:07:00,374 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:07:15,438 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:07:30,480 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:07:45,536 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:08:00,587 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:08:15,645 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:08:30,698 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:08:45,756 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:09:00,838 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:09:15,886 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:09:30,951 ec2-spot-converter - Waiting for instance to stop... (current state=running)
[INFO] 2021-07-03 01:09:46,010 ec2-spot-converter - Waiting for instance to stop... (current state=running)
```

However, the first time I ran this, the instance stopped at step 6, but no more steps were executed (I waited for more than 10 minutes). Now when I am trying it (tried it multiple times), it's getting stuck at step 6 as above. 

Any idea why this might be happening? I am new to this, so bear with me if my question sounds noob.

**Comment by jcjorel** [comment]
Hi, you may have found a bug…
Can you start the command again please ? It is likely to cycle again waiting for instance stop but, this time go to the AWS console and issue a stop on the instance interactively. Tell me if it allows to proceed the conversion.

**Comment by shalabhsingh** [comment]
Hi. I did as you said and the SSH connection to my instance closed when I stopped the instance interactively

**Comment by jcjorel** [comment]
Ok. I may understand. The tool needs to be run from another instance than the one you are converting. Can you try this?

**Comment by jcjorel** [comment]
Any update?

**Comment by jcjorel** [comment]
Closing issue as no reply from issuer.

**Comment by shalabhsingh** [comment]
Sorry for the late response. 
So I tried it from a different from another instance. This time too, the process was stuck at Step 6 for around 150 seconds, but I manually stopped the instance and it worked. However the process got stuck at a later stage-

```
[INFO] 2021-07-10 08:57:26,097 ec2-spot-converter - [STEP 25/26] Waiting for instance to be at expected states in target groups... => SKIPPED! Need '--wait-for-tg-states' argument.
[INFO] 2021-07-10 08:57:26,097 ec2-spot-converter - [STEP 26/26] Deregister image... => SKIPPED! Need '--delete-ami' argument.
[INFO] 2021-07-10 08:57:26,098 ec2-spot-converter - Conversion successful! New instance id: <id>, ElapsedTime: 634577 seconds
Traceback (most recent call last):
  File "./ec2-spot-converter", line 1558, in <module>
    sys.exit(main(sys.argv))
  File "./ec2-spot-converter", line 1549, in main
    review_conversion_results()
  File "./ec2-spot-converter", line 1343, in review_conversion_results
    initial_instance_state["Tags"]                = sorted(initial_instance_state["Tags"], key=lambda t: t["Key"])
KeyError: 'Tags'
```

In the ec2 instances console, I can see a new instance created with `Terminated` state and my original instance (which was stopped at step 6), again in `Running` state.

How should I proceed?

**Comment by jcjorel** [comment]
Hi,
According to where the exception happened, the conversion should have succeeded. The new running instances is the newly created. Have a look to instance id, the one in 'Terminated' should be the one supplied on the command line. Can you confirm it?

But, indeed, you found a bug that causes an Exception with trying to display the VimDiff at end of a successful conversion if the original instance has no tag. I just fixed it and release v0.12.1 for general availability. With this version, you should have no more this exception.

**Comment by shalabhsingh** [comment]
Yes, the terminated instance was the one supplied on command line. 

Thanks for the help.
